### PR TITLE
Delete and list by date/time

### PR DESCRIPTION
**Pull request by yuchenlichuck**
#106 
#105

### CODE REVIEW

**Comment by yuchenlichuck** [comment]
I have also finished in ug. I have written delete by category and by category+date+time since I haven't seen these in the lastest version #137

**Comment by rsanchez-macias** [comment]
I think we should rename some methods and variables before merging. For example, we could change "ListAll" for something like "DeleteAll"

**Comment by rsanchez-macias** [comment]
And then another would be ```getListByCategory()``` to maybe ```deleteByCategory()```

**Comment by yuchenlichuck** [comment]
You are definitely right! Sorry for forgetting change the name of the methods

**Comment by Bencotti** [comment]
Please stop merging your own PR, if it's taking some time to get it approve you can assign us to review. There are portions of your code that need revision.

File: src/main/java/seedu/command/DeleteCommand.java [hunk 1]
```diff
@@ -140,93 +132,80 @@ private String getListByDateCategory(String date, String time, String category)
...
+                        if (localTimes.get(2 * j).isBefore(endTimes.get(k))
+                                && localTimes.get(2 * j + 1).isAfter(startTimes.get(k))) {
+                            Task removedTask = taskList.deleteTask(i);
+                            storage.overwriteFile(taskList.getList());
+                            assert removedTask != null : "Removed-task is null";
+                            feedback += formatSuccessFeedback(removedTask) + "\n";
+                            i--;
+                            break label1;
                         }
                     }
```
Comment (Bencotti) [inline src/main/java/seedu/command/DeleteCommand.java:195]: Should try to abstract some of the code out to make it less nested

**Review by JTWang2000 — COMMENTED** [review]
image
Could you please specify here?

**Comment by JTWang2000** [comment]
Line 135,136,137 is list date; list date/time usage why you add it here? Just to increase the length or code?
image
Please check the UG of list/delete carefully and make sure it matches your implementation.

**Pull request merged**